src/query/query_executor.c

```diff
@@ -16070,11 +16070,17 @@ qexec_execute_cte (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STATE * xasl_
 
 	  if (ent == NULL || list_cache_entry_p == NULL || list_id == NULL)
```

Comment:
trivial,
We can get the same result by just checking `list_id`.
```suggestion
	  if (list_id == NULL)
```

Reply:
good point. thanks.